third_party/packages/ansi_up/lib/ansi_up.dart

```diff
@@ -2,76 +2,306 @@
+      //  this._buffer = this._buffer.slice(rpos);
+      //  return pkt;
+      //}
+    }
+    return pkt;
+  }
+
+  void _appendBuffer(String text) {
+    _buffer = _buffer + text;
+  }
+
+  void _processAnsi(TextPacket textPacket) {
+    final List<String> sgrCmds = textPacket.text.split(';');
+    while (sgrCmds.isNotEmpty) {
+      final sgrCmdStr = sgrCmds.removeAt(0);
+      final num = int.tryParse(sgrCmdStr, radix: 10);
+      if (num == null || num == 0) {
+        fg = bg = null;
+        bold = false;
+      } else if (num == 1) {
+        bold = true;
+      } else if (num == 22) {
+        bold = false;
+      } else if (num == 39) {
+        fg = null;
+      } else if (num == 49) {
+        bg = null;
+      } else if ((num >= 30) && (num < 38)) {
+        fg = ansiColors[0][(num - 30)];
+      } else if ((num >= 40) && (num < 48)) {
+        bg = ansiColors[0][(num - 40)];
+      } else if ((num >= 90) && (num < 98)) {
+        fg = ansiColors[1][(num - 90)];
+      } else if ((num >= 100) && (num < 108)) {
+        bg = ansiColors[1][(num - 100)];
+      } else if (num == 38 || num == 48) {
+        if (sgrCmds.isNotEmpty) {
+          final bool isForeground = num == 38;
+          final String modeCmd = sgrCmds.removeAt(0);
+          if (modeCmd == '5' && sgrCmds.isNotEmpty) {
+            final paletteIndex = int.tryParse(sgrCmds.removeAt(0), radix: 10);
+            if (paletteIndex >= 0 && paletteIndex <= 255) {
+              if (isForeground)
+                fg = palette256[paletteIndex];
+              else
+                bg = palette256[paletteIndex];
+            }
+          }
+          if (modeCmd == '2' && sgrCmds.length > 2) {
+            final int r = int.tryParse(sgrCmds.removeAt(0), radix: 10);
+            final int g = int.tryParse(sgrCmds.removeAt(0), radix: 10);
+            final int b = int.tryParse(sgrCmds.removeAt(0), radix: 10);
+            if ((r >= 0 && r <= 255) &&
+                (g >= 0 && g <= 255) &&
+                (b >= 0 && b <= 255)) {
+              var c = AnsiUpColor(rgb: [r, g, b], class_name: 'truecolor');
+              if (isForeground)
+                fg = c;
+              else
+                bg = c;
+            }
+          }
+        }
+      }
+    }
+  }
+
+  TextWithAttr _withState(TextPacket packet) {
+    return TextWithAttr(bold: bold, fg: fg, bg: bg, text: packet.text);
+  }
 }
 
-@JS()
 class TextWithAttr {
-  external AU_Color get fg;
-  external AU_Color get bg;
-  external bool get bold;
-  external String get text;
+  TextWithAttr({this.fg, this.bg, this.bold, this.text});
+
+  AnsiUpColor fg;
+  AnsiUpColor bg;
+  bool bold;
+  String text;
 }
 
-@JS()
-@anonymous
-class AU_Color {
-  external List<int> get rgb;
-  external String get class_name;
+class AnsiUpColor {
+  AnsiUpColor({this.rgb, this.class_name});
+
+  List<int> rgb;
+  String class_name;
 }
 
-class PacketKind {
-  static final int EOS = 0;
-  static final int Text = 1;
+mixin PacketKind {
```

Comment:
why is this a mixin?

Reply:
now that this code is all Dart you could as a quick followup turn this into an actual Dart enum
`enum PacketKind { EOS, Text, ...}`

Reply:
It was originally a class so the automatic linting suggested a mixin instead 😅 Enum makes a lot more sense though!